dozer-api/src/rest/tests/routes.rs

```diff
@@ -257,3 +259,112 @@ async fn path_collision_test() {
+    let endpoint = test_utils::get_endpoint();
+    let cache_manager = test_utils::initialize_cache(&endpoint.name, None);
+    let api_server = ApiServer::create_app_entry(
+        None,
+        CorsOptions::Permissive,
+        vec![Arc::new(
+            CacheEndpoint::open(&*cache_manager, Default::default(), endpoint.clone()).unwrap(),
+        )],
+        Default::default(),
+    );
+    let app = actix_web::test::init_service(api_server).await;
+
+    let req = actix_web::test::TestRequest::post()
+        .insert_header(ContentType::plaintext())
+        .uri(&format!("{path}/query", path = endpoint.path))
+        .to_request();
+    let res = actix_web::test::call_service(&app, req).await;
+
+    // Assert bad request
+    assert_eq!(res.status(), StatusCode::BAD_REQUEST);
+}
+#[actix_web::test]
+async fn test_empty_content_type() {
+    let endpoint = test_utils::get_endpoint();
+    let cache_manager = test_utils::initialize_cache(&endpoint.name, None);
+    let api_server = ApiServer::create_app_entry(
+        None,
+        CorsOptions::Permissive,
+        vec![Arc::new(
+            CacheEndpoint::open(&*cache_manager, Default::default(), endpoint.clone()).unwrap(),
+        )],
+        Default::default(),
+    );
+    let app = actix_web::test::init_service(api_server).await;
+
+    let (count, records) = count_and_query(&endpoint.path, &app, None).await;
+    assert_eq!(count, 52);
+    assert_eq!(records.len(), 50);
+}
+
+#[actix_web::test]
+async fn test_correct_content_type() {
+    let endpoint = test_utils::get_endpoint();
+    let cache_manager = test_utils::initialize_cache(&endpoint.name, None);
+    let api_server = ApiServer::create_app_entry(
+        None,
+        CorsOptions::Permissive,
+        vec![Arc::new(
+            CacheEndpoint::open(&*cache_manager, Default::default(), endpoint.clone()).unwrap(),
+        )],
+        Default::default(),
+    );
+    let app = actix_web::test::init_service(api_server).await;
+
+    let req_query = actix_web::test::TestRequest::post()
+        .uri(&format!("{path}/query", path = endpoint.path))
+        .insert_header(ContentType::json())
+        .to_request();
+    let res_query = actix_web::test::call_service(&app, req_query).await;
+
+    // Assert success
+    assert!(res_query.status().is_success());
+    let body1: Value = actix_web::test::read_body_json(res_query).await;
+    let records = body1.as_array().unwrap().to_vec();
+    assert_eq!(records.len(), 50);
+
+    let req_count = actix_web::test::TestRequest::post()
+        .uri(&format!("{path}/count", path = endpoint.path))
+        .insert_header(ContentType::json())
+        .to_request();
+    let res_count = actix_web::test::call_service(&app, req_count).await;
+
+    // Assert success
+    assert!(res_count.status().is_success());
+    let body: Value = actix_web::test::read_body_json(res_count).await;
+    let count = body.as_u64().unwrap();
+    assert_eq!(count, 52);
```

Comment:
We can add a `content_type` parameter to `count_and_query` to avoid code duplication?